### PR TITLE
Replace assertion with ValueError exceptions in run_image_captioning_flax.py

### PR DESCRIPTION
**Pull request by katiele47**
# What does this PR do?

Replaces 4 `assert` with `ValueError` exeception in run_image_captioning_flax.py.
Co-author: @AdiaWu

Related to [#12789](https://github.com/huggingface/transformers/issues/12789).

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@sgugger

### CODE REVIEW

**Comment by sgugger** [comment]
Thanks, but the issue was intended for library code, not examples. Let's see if @sanchit-gandhi doesn't mind!

File: examples/flax/image-captioning/run_image_captioning_flax.py [hunk 1]
```diff
@@ -502,7 +504,8 @@ def main():
     # Get the column names for input/target.
     dataset_columns = image_captioning_name_mapping.get(data_args.dataset_name, None)
     if data_args.image_column is None:
-        assert dataset_columns is not None
+        if dataset_columns is None:
+            raise ValueError("Columns of dataset should not be None")
```
Comment (sanchit-gandhi) [inline examples/flax/image-captioning/run_image_captioning_flax.py:508]: `datasets_columns` is None if `data_args.dataset_name` is not in `image_captioning_name_mapping`:

https://github.com/huggingface/transformers/blob/f3a1efd1cf2e8b2aadec4a41d8bc8ef470de0275/examples/flax/image-captioning/run_image_captioning_flax.py#LL503

We should try and make the description for this ValueError reveal this problem! Could we state that `data_args.dataset_name` is not in `image_captioning_name_mapping` in the error message?

```suggestion
            raise ValueError(f"`--dataset_name` {data_args.dataset_name} not found in dataset '{data_args.dataset_name}'. Make sure to set `--dataset_name` to the correct dataset name, one of {', '.join(image_captioning_name_mapping.keys())}.")
```

(might need to run "make style" after incorporating these changes!)

File: examples/flax/image-captioning/run_image_captioning_flax.py [hunk 2]
```diff
@@ -511,7 +514,8 @@ def main():
                 f"--image_column' value '{data_args.image_column}' needs to be one of: {', '.join(column_names)}"
             )
     if data_args.caption_column is None:
-        assert dataset_columns is not None
+        if dataset_columns is None:
+            raise ValueError("Columns of dataset should not be None")
```
Comment (sanchit-gandhi) [inline examples/flax/image-captioning/run_image_captioning_flax.py:518]: Same here please!

```suggestion
            raise ValueError(f"`--dataset_name` {data_args.dataset_name} not found in dataset '{data_args.dataset_name}'. Make sure to set `--dataset_name` to the correct dataset name, one of {', '.join(image_captioning_name_mapping.keys())}.")
```

**Review by sanchit-gandhi — COMMENTED** [review]
Thanks @katiele47! Could we make the second set of `ValueError`'s more descriptive of what's going wrong? (see below)

**Comment by katiele47** [comment]
Thank you for your suggestions @sanchit-gandhi! @sgugger ahh I should have known that. I'll update the library code next time.

**Comment by sanchit-gandhi** [comment]
Hey @katiele47, sorry for the late reply! 

Could you check that you've installed Transformers form source? https://huggingface.co/docs/transformers/pr_checks#checks-on-a-pull-request

You can make sure you've installed the dev version of transformers by uninstalling transformers then pip installing from within the transformers repo:
```
pip uninstall transformers
pip install -e .[dev]
```

You should then be able to run `make style` for code quality fixes!

**Comment by JuheonChu** [comment]
Hello @katiele47, in my case, 

I used 

- `pip install black`
- `black [your filepath]`

and this solved passing the CircleCi code quality tests.

**Review by sanchit-gandhi — APPROVED** [review]
Thanks for iterating on this @katiele47!

**Comment by katiele47** [comment]
Thank you so much for reviewing my change @sanchit-gandhi!

**Pull request merged**